Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp

```diff
@@ -1095,13 +1100,48 @@ FTransform ARRAIRobotROSController::GetOrigin()
     }
 }
 
+bool ARRAIRobotROSController::UpdateGoalSequenceIndex(FTransform& OutGoal)
+{
+    if (GoalSequence.Num() > 0)
+    {
+        GoalIndex++;
+        if (GoalSequence.Num() <= GoalIndex)
+        {
+            GoalIndex = 0;
+        }
+        OutGoal = GoalSequence[GoalIndex];
+        return true;
+    }
+    return false;
+}
+
+bool ARRAIRobotROSController::RandomUpdateGoalSequenceIndex(FTransform& OutGoal)
+{
+    if (GoalSequence.Num() > 0)
+    {
+        if (GoalSequence.Num() > 2)
+        {
+            int temp = FMath::RandRange(0, GoalSequence.Num() - 1);
+            while (temp == GoalIndex)
+            {
+                temp = FMath::RandRange(0, GoalSequence.Num() - 1);
+            }
+            GoalIndex = temp;
+            OutGoal = GoalSequence[GoalIndex];
+            return true;
+        }
+    }
+    return false;
+}
+
 void ARRAIRobotROSController::ModeUpdate()
 {
     if (Mode != ERRAIRobotMode::MANUAL)
     {
         if (NavStatus == ERRAIRobotNavStatus::IDLE)
         {
             FMoveCompleteCallback onSuccess, onFail;    // dummy
+            FTransform dummy;

```

Comment:
It seems dummy is initialized after but never used

Reply:
it is used  at line 1155 and 1158

Reply:
these lines initialize dummy but then it is not used?

Reply:
Those are mainly used by Blueprint to get output transform directly.